src/libraries/Microsoft.PowerFx.Core/Entities/External/IExternalTabularDataSource.cs

```diff
@@ -11,6 +11,13 @@ internal interface IExternalTabularDataSource : IExternalDataSource, IDisplayMap
     {
         TabularDataQueryOptions QueryOptions { get; }
 
+        /// <summary>
+        /// Some data sources (like Dataverse) may return a cached value for
+        /// the number of rows (calls to CountRows) instead of always retrieving
+        /// the latest count.
+        /// </summary>
+        bool HasCachedCountRows { get; }
```

Comment:
>bool HasCachedCountRows { get; } [](http://example.com/codeflow?start=8&length=32)

Why not at `IExternalDataSource`?

Reply:
CountRows can only be applied to tabular data sources. An API data source such as Office365Users, which has lots of operations, doesn't have the concept of a count.